manifests/consul_dns.pp

```diff
@@ -120,11 +123,16 @@
     }
   }
 
+  $dnsmasq_client_addr = $client_addr ? {
+    '0.0.0.0' => $::ipaddress_lo,
+    default   => $client_addr,
+  }
+
   # Dnsmasq
   # -------
   $dnsmasq_base_opts = {
     'cache-size' => '0',
-    'server'     => "/${domain}/${client_addr}#8600",
+    'server'     => "/${domain}/${dnsmasq_client_addr}#8600",
     'address'    => "/${dnsmasq_host_alias}/${advertise_addr}",
```

Comment:
We did. I'd prefer to do it in a new PR because I haven't tested the behaviour change here.